### ISSUE TITLE
Random Connection reset error After upgrading agent from 3.x to 4.x

### ISSUE THREAD

**Issue by naveen-tenmiles** [elastic/apm-agent-python#437]
**Describe the bug**: ...

Server timeout from django wsgi api, This issue happens once i upgraded the python client from 

**To Reproduce**

1. APM server 6.6.2
2. Apm client versions (4.1.0, 4.2.0,4.2.1)

**Expected behavior**: No errors

**Environment (please complete the following information)**
- OS: Linux ubuntu 18.04
- Python version: 2.7.15
- Framework and version [e.g. Django 2.1]: Django 1.11
- APM Server version: 6.6.2
- Agent version: 4.1.0, 4.2.0,4.2.1

**Additional context**

```
ValueError: close timed out
  File "elasticapm/utils/compat.py", line 65, in uwsgi_atexit
    func()
  File "elasticapm/base.py", line 236, in close
    self._transport.close()
  File "elasticapm/transport/base.py", line 290, in close
    super(AsyncTransport, self).close()
  File "elasticapm/transport/base.py", line 238, in close
    raise ValueError("close timed out")
```

After upgrading agent to v4.2.1 below error also occurs.

```
RuntimeError: cannot join thread before it is started
  File "django/http/response.py", line 264, in close
    signals.request_finished.send(sender=self._handler_class)
  File "django/dispatch/dispatcher.py", line 193, in send
    for receiver in self._live_receivers(sender)
  File "elasticapm/contrib/django/apps.py", line 50, in <lambda>
    lambda sender, **kwargs: client.end_transaction() if _should_start_transaction(client) else None,
  File "elasticapm/base.py", line 260, in end_transaction
    self._collect_transactions()
  File "elasticapm/base.py", line 298, in _collect_transactions
    self._stop_send_timer()
  File "elasticapm/base.py", line 323, in _stop_send_timer
    self._send_timer.join()
  File "python2.7/threading.py", line 929, in join
    raise RuntimeError("cannot join thread before it is started")
```

**Note:** I tried increasing read/write timeout to 45s but doesn't help.

```
  # Maximum permitted duration for reading an entire request.
  read_timeout: 45s
  # Maximum permitted duration for writing a response.
  write_timeout: 45s
```

**Comment by beniwohli** [comment]
Hi @naveen-tenmiles 

Sorry for the long reply time :( can you post your full uwsgi configuration? There seem to be some issues with getting our background thread started correctly on uwsgi, which can lead to this issue.

**Comment by naveen-tenmiles** [comment]
Hi @beniwohli ,

i have attached my uwsgi config below.

```
[uwsgi]
file = code/app/wsgi.py
virtualenv = virtual_env_v2/
master = true
http-socket = :9002
chmod-socket = 666
chdir = /srv/testing
uid = myuser
gid = myuser
stats = /tmp/stats_v2.socket
enable-threads = true
single-interpreter = true
# max concurrency level (number of cpu cores * 4)
threads = %(%k * 4)
cheaper-algo = spare
# minimum workers at all times
cheaper = 2
# no of workers in startup
cheaper-initial = 2
# max no of workers at all times (number of cpu cores * 2)
workers = %(%k * 2)
cheaper-step = 3
# busyness check window (in seconds)
cheaper-overload = 30
# timeout
harakiri = 60
# reload a worker after serving 500 requests
max-requests = 500
# reload a worker if its physical unshared memory is higher than the specified value (in mb)
reload-on-rss = 1024
reload-on-exception-value = connection already closed
buffer-size = 10240
# to server static files from uwsgi
static-map = /media=/srv/testing/code/website/media
static-expires = /* 7776000
offload-threads = %k
# for handling ioerrors (https://github.com/getsentry/raven-python/issues/732)
ignore-sigpipe = true
ignore-write-errors = true
disable-write-exception = true
```

**Comment by beniwohli** [comment]
@naveen-tenmiles we recently released version 4.2.2 of the agent which should take care of the threading issue you saw. I'll close this for now, please feel free to re-open if the problem still persists after upgrading.
